atc/db/worker_factory.go

```diff
@@ -397,10 +397,13 @@ func saveWorker(tx Tx, atcWorker atc.Worker, teamID *int, ttl time.Duration, con
 		workerVersion = &atcWorker.Version
 	}
 
+	activeTasks := 0 // When a worker starts up by definition has no active tasks.
```

Comment:
When calling `saveWorker`, it is possible for the worker to already exist in the DB. This can happen in a few ways:
1) when there is a network blip and the worker is marked as stalled and then re-registers
2) when the worker rebalances to another ATC after some time.

In these cases, we would expect that active_tasks should remain the same as they were previously. active_tasks should only be set to 0 for new/recreated workers.